src/swupd/manifest.go

```diff
@@ -0,0 +1,118 @@
+package swupd
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+)
+
+// ManifestHeader contains metadata for the manifest
+type ManifestHeader struct {
+	Format      int
+	Version     int32
+	Previous    int32
+	FileCount   int32
+	TimeStamp   time.Time
+	ContentSize int64
+	Includes    []Manifest
+}
+
+// Manifest represents a bundle or list of bundles (MoM)
+type Manifest struct {
+	Name         string
+	Header       ManifestHeader
+	Files        []File
+	AddedCount   uint32
+	RemovedCount uint32
+	ChangedCount uint32
```

Comment:
Minor: we talked about getting rid of those Counters, but it is fine if we do later on.

Reply:
Oops, yes.

Reply:
If we have them instead as arrays of pointers to interned files (added files. removed files etc), then we always have len() which is a VERY cheap operation to get this data anyhow.

I think the point I made about this being the _internal_ (in memory) representation rather than the on disk version is worth repeating.

Reply:
Except we still need to check more than just the len in the case where then length stays the same but we have several changed files.

Reply:
My suggestion: remove the counters, don't add anything else. When the code that actually uses that information (added/etc) is added, it adds the necessary information (e.g. arrays of pointers, etc).